HCSP.js

```diff
@@ -0,0 +1,228 @@
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+// Usage of Zotero.Utilities.xpath() function. Exist only since Zotero 2.1
+// https://www.zotero.org/support/dev/client_coding/changes_in_zotero_2.1#xpath_utility_functions
+// Then minimal version of Zotero needed is 2.1
+
+// TODO: Use camelStyle for variables to be consistent with Zotero coding style
+
+function detectWeb(doc, url) {
+    /* The URLs have upper or lower case on the website
+    depending the way you access to a page */
+    if (url.match("[Ee]xplore.cgi/avisrapportsdomaine")) {
+        return 'report';
+    } else if (url.match("[Ee]xplore.cgi/avisrapports")) {
+        return 'multiple';
+    }
+}
+
+function doWeb(doc, url) {
+
+    // Store articles URLs in a array
+    var articles = [];
+
+    if (detectWeb(doc, url) == "multiple") {
+
+        // Titles and URLs of each item are on the third column of the only table
+        // This column is the only one with links (<a>)
+        var xpath_titles = '//tbody/tr/td/a';
+        var titles = ZU.xpath(doc, xpath_titles);
+        
+        // Put all the titles and their URLs in the items object
+        // NOTE: "for of loop" are quite new and works only with newest browsers
+        // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of
+        var allItems = {};
+        for (var nextTitle of titles) {
+            allItems[nextTitle.href] = nextTitle.textContent;
+        }
+        
+        // Let the user choose the items he wants
+        var itemsSelected;
+        itemsSelected = Zotero.selectItems(allItems);
+
+        // Save these articles
+        for (var one_item in itemsSelected) {
+            articles.push(one_item)
+        }
+
+    } else {
+        // Page of the article : get the URL
+        articles = [url]
+    }
+
+    Zotero.Utilities.processDocuments(articles, scrape, function(){Zotero.done();});
+    Zotero.wait();
+}
+
+function scrape(doc, url) {
+
+    var namespace = doc.documentElement.namespaceURI;
+    // Resolver for XPaths
+
+    // init a new Zotero's item
+    var newItem = new Zotero.Item("report");
+
+    /*---------- Constants items ----------*/
+    newItem.publicationTitle = "Rapport de l'HCSP";
+    newItem.language = "fr";
+    newItem.institution = "Haut Conseil de la Santé Publique";
+    newItem.place = "Paris";
+
+    // Creator is the institution (HCSP)
+    // Names of the author are inside the PDFs (if there is one)
+    // TODO: Find a way to have a single field instead of two
+    var creator = ZU.cleanAuthor('HCSP', 'author', false);
+    newItem.creators.push(creator);
```

Comment:
so ```ZU.cleanAuthor()``` just generates a Zotero creator object -- the same one you can see down in the translator test -- from a string (and some optional arguments like the creator type). But instead, you can just write that object out yourself, which is what I suggested to you here. Among other things, that allows you to just specify the fieldMode rather than adding it later on.